api/grpc/server/server.go

```diff
@@ -287,3 +287,20 @@ func (s *apiServer) GetStats(r *types.StatsRequest, stream types.API_GetStatsSer
 	}
 	return nil
 }
+
+func (s *apiServer) PullStats(ctx context.Context, r *types.PullStatsRequest) (*types.PullStatsResponse, error) {
+	res := &types.PullStatsResponse{
+		Stats: make(map[string]*types.Stats),
+	}
+	for _, id := range r.Ids {
+		e := supervisor.NewEvent(supervisor.StatsEventType)
+		e.ID = id
+		s.sv.SendEvent(e)
+		if err := <-e.Err; err != nil {
+			res.Errors = append(res.Errors, &types.StatError{Id: id, Error: err.Error()})
```

Comment:
Like on the first failure? I thought it might be client task to handle errors. Skip stats with error or error out.